packages/volto/src/reducers/favorites/favoritesReducer.js

```diff
@@ -0,0 +1,115 @@
+import {
+  ADD_TO_FAVORITES,
+  REMOVE_FROM_FAVORITES,
+  LOAD_FAVORITES,
+} from '../../actions/favorites/favorites';
+
+const STORAGE_KEY = 'volto_favorite_blocks';
+
+const extractPreviewData = (blockData) => {
+  if (!blockData) return null;
+
+  switch (blockData['@type']) {
+    case 'image':
+      return {
+        type: 'single',
+        images: [
+          {
+            url: blockData.url,
+            alt: blockData.alt || 'Image preview',
+          },
+        ],
+      };
+    case 'imagegallery':
+      return {
+        type: 'gallery',
+        images: (blockData.images || []).map((img) => ({
+          url: img.url || img['@id'],
+          alt: img.alt || 'Gallery image',
+        })),
+      };
+    case 'video':
+      return {
+        type: 'video',
+        url: blockData.url,
+      };
+    case 'slate':
+      return {
+        type: 'text',
+        content: blockData.value?.[0]?.children?.[0]?.text || '',
+      };
+    default:
+      return {
+        type: 'default',
+        content: blockData['@type'],
+      };
```

Comment:
maybe you can allow developers to extend this reducer from config.blocks, for custom new blocks